libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -394,13 +393,13 @@ internal class MapboxTripSession(
         unconditionalStatusPollingJob?.cancel()
         this.rawLocation = rawLocation
         locationObservers.forEach { it.onRawLocationChanged(rawLocation) }
-        navigatorJobController.scope.launch {
+        ioJobController.scope.launch {
```

Comment:
yeah updateDataFromNavigatorStatus can be a suspending function. where it updates callbacks on the main dispatcher and updates all the data on the navigator dispatcher